tests/server/dataSource-tests.js

```diff
@@ -0,0 +1,606 @@
+                }],
+                reject: "sjrk.test.storyTelling.server.dataSource.tester.assertMissingError",
+                rejectArgs: [
+                    "Error thrown for invalid story id",
+                    "{arguments}.0",
+                    {
+                        storyId: "invalid_story_id",
+                        rev: "{that}.rev",
+                        method: "DELETE"
+                    }
+                ]
+            }, {
+                task: "{deleteStoryDataSource}.set",
+                args: [{
+                    directStoryId: "story1",
+                    directRevisionId: "{that}.rev"
+                }],
+                resolve: "sjrk.test.storyTelling.server.dataSource.tester.assertResponse",
+                resolveArgs: [
+                    "Story should have been removed",
+                    {
+                        "ok": true,
+                        "id": "story1"
+                    },
+                    "{arguments}.0"
+                ]
+            }, {
+                task: "{storyDataSource}.get",
+                args: [{directStoryId: "story1"}],
+                reject: "sjrk.test.storyTelling.server.dataSource.tester.assertMissingError",
+                rejectArgs: [
+                    "Error thrown for attempting to access a deleted story",
+                    "{arguments}.0",
+                    {storyId: "story1"}
+                ]
+            }]
+        }]
+    }]
+});
+
+/**
+ * A wrapper around `jqUnit.assertDeepEq` except it strips `_rev` out of the actual response as this is generated value.
+ * The remaining properties are compared with `jqUnit.assertDeepEq`.
+ *
+ * @param {String} msg - The assertion message
+ * @param {Object} expected - The expected value to compare against the `actual` value.
+ * @param {Object} actual - The actual value, minus `_rev` property, compared against the `expected` value.
+ */
+sjrk.test.storyTelling.server.dataSource.tester.assertResponse = function (msg, expected, actual) {
+    actual = fluid.filterKeys(actual, "_rev", true);
+    jqUnit.assertDeepEq(msg, expected, actual);
+};
+
+/**
+ * A wrapper around `jqUnit.assertDeepEq`, includes construction of the expected error message
+ *
+ * @param {String} msg - The assertion message
+ * @param {Object} actual - The actual error returned
+ * @param {Object} [options] - (optional) values to be interpolated into the error message
+ * @param {String} [options.method] - (optional) the HTTP method to interpolate into error message; default "GET"
+ * @param {String} [options.storyId] - (optional) the storyId to interpolate into the error message; default ""
+ * @param {String} [options.rev] - (optional) the rev to interpolate into the error message
+ */
+sjrk.test.storyTelling.server.dataSource.tester.assertMissingError = function (msg, actual, options) {
+    var tokens = fluid.merge("replace", {}, {method: "GET", storyID: ""}, options || {});
+    var expected = {
```

Comment:
Would the expected values be better defined outside of this function and passed in?